diff --git a/node_modules/@aries-framework/askar/build/wallet/AskarWallet.js b/node_modules/@aries-framework/askar/build/wallet/AskarWallet.js
index 2791f91..0c9e653 100644
--- a/node_modules/@aries-framework/askar/build/wallet/AskarWallet.js
+++ b/node_modules/@aries-framework/askar/build/wallet/AskarWallet.js
@@ -291,6 +291,7 @@ let AskarWallet = class AskarWallet {
             this._store = undefined;
         }
         catch (error) {
+            this.fileSystem.delete(destinationPath)
             const errorMessage = `Error closing wallet': ${error.message}`;
             this.logger.error(errorMessage, {
                 error,
