diff --git a/node_modules/@credo-ts/push-notifications/build/fcm/PushNotificationsFcmService.js b/node_modules/@credo-ts/push-notifications/build/fcm/PushNotificationsFcmService.js
index ac9bf8e..497da31 100644
--- a/node_modules/@credo-ts/push-notifications/build/fcm/PushNotificationsFcmService.js
+++ b/node_modules/@credo-ts/push-notifications/build/fcm/PushNotificationsFcmService.js
@@ -30,6 +30,7 @@ let PushNotificationsFcmService = class PushNotificationsFcmService {
             threadId,
             deviceToken: deviceInfo.deviceToken,
             devicePlatform: deviceInfo.devicePlatform,
+            clientCode: deviceInfo.clientCode,
         });
     }
     processSetDeviceInfo(messageContext) {
diff --git a/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmDeviceInfoMessage.d.ts b/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmDeviceInfoMessage.d.ts
index 65ac2cf..533cac7 100644
--- a/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmDeviceInfoMessage.d.ts
+++ b/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmDeviceInfoMessage.d.ts
@@ -14,6 +14,7 @@ export declare class PushNotificationsFcmDeviceInfoMessage extends AgentMessage
     constructor(options: PushNotificationsFcmDeviceInfoOptions);
     deviceToken: string | null;
     devicePlatform: string | null;
+    clientCode: string | null;
     readonly type: string;
     static readonly type: import("@credo-ts/core/build/utils/messageType").ParsedMessageType;
 }
diff --git a/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmDeviceInfoMessage.js b/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmDeviceInfoMessage.js
index c1ab34d..b18c62b 100644
--- a/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmDeviceInfoMessage.js
+++ b/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmDeviceInfoMessage.js
@@ -46,6 +46,12 @@ __decorate([
     (0, class_validator_1.ValidateIf)((object, value) => value !== null),
     __metadata("design:type", Object)
 ], PushNotificationsFcmDeviceInfoMessage.prototype, "devicePlatform", void 0);
+__decorate([
+    (0, class_transformer_1.Expose)({ name: 'client_code' }),
+    (0, class_validator_1.IsString)(),
+    (0, class_validator_1.ValidateIf)((object, value) => value !== null),
+    __metadata("design:type", Object)
+], PushNotificationsFcmDeviceInfoMessage.prototype, "clientCode", void 0);
 __decorate([
     (0, core_1.IsValidMessageType)(PushNotificationsFcmDeviceInfoMessage.type),
     __metadata("design:type", Object)
diff --git a/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmSetDeviceInfoMessage.d.ts b/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmSetDeviceInfoMessage.d.ts
index f82cf0f..c35c836 100644
--- a/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmSetDeviceInfoMessage.d.ts
+++ b/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmSetDeviceInfoMessage.d.ts
@@ -12,6 +12,7 @@ export declare class PushNotificationsFcmSetDeviceInfoMessage extends AgentMessa
     constructor(options: PushNotificationsFcmSetDeviceInfoOptions);
     deviceToken: string | null;
     devicePlatform: string | null;
+    clientCode: string | null;
     readonly type: string;
     static readonly type: import("@credo-ts/core/build/utils/messageType").ParsedMessageType;
 }
diff --git a/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmSetDeviceInfoMessage.js b/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmSetDeviceInfoMessage.js
index 0fd2c32..5e66fae 100644
--- a/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmSetDeviceInfoMessage.js
+++ b/node_modules/@credo-ts/push-notifications/build/fcm/messages/PushNotificationsFcmSetDeviceInfoMessage.js
@@ -27,6 +27,7 @@ class PushNotificationsFcmSetDeviceInfoMessage extends core_1.AgentMessage {
             this.id = (_a = options.id) !== null && _a !== void 0 ? _a : this.generateId();
             this.deviceToken = options.deviceToken;
             this.devicePlatform = options.devicePlatform;
+            this.clientCode = options.clientCode;
         }
     }
 }
@@ -44,6 +45,12 @@ __decorate([
     (0, class_validator_1.ValidateIf)((object, value) => value !== null),
     __metadata("design:type", Object)
 ], PushNotificationsFcmSetDeviceInfoMessage.prototype, "devicePlatform", void 0);
+__decorate([
+    (0, class_transformer_1.Expose)({ name: 'client_code' }),
+    (0, class_validator_1.IsString)(),
+    (0, class_validator_1.ValidateIf)((object, value) => value !== null),
+    __metadata("design:type", Object)
+], PushNotificationsFcmSetDeviceInfoMessage.prototype, "clientCode", void 0);
 __decorate([
     (0, core_1.IsValidMessageType)(PushNotificationsFcmSetDeviceInfoMessage.type),
     __metadata("design:type", Object)
diff --git a/node_modules/@credo-ts/push-notifications/build/fcm/models/FcmDeviceInfo.d.ts b/node_modules/@credo-ts/push-notifications/build/fcm/models/FcmDeviceInfo.d.ts
index ab8c284..5186d39 100644
--- a/node_modules/@credo-ts/push-notifications/build/fcm/models/FcmDeviceInfo.d.ts
+++ b/node_modules/@credo-ts/push-notifications/build/fcm/models/FcmDeviceInfo.d.ts
@@ -1,4 +1,5 @@
 export type FcmDeviceInfo = {
     deviceToken: string | null;
     devicePlatform: string | null;
+    clientCode: string | null;
 };