diff --git a/node_modules/@credo-ts/indy-vdr/build/anoncreds/IndyVdrAnonCredsRegistry.js b/node_modules/@credo-ts/indy-vdr/build/anoncreds/IndyVdrAnonCredsRegistry.js
index 30ded2c..5928d48 100644
--- a/node_modules/@credo-ts/indy-vdr/build/anoncreds/IndyVdrAnonCredsRegistry.js
+++ b/node_modules/@credo-ts/indy-vdr/build/anoncreds/IndyVdrAnonCredsRegistry.js
@@ -182,7 +182,7 @@ class IndyVdrAnonCredsRegistry {
             }
             // Format the schema id based on the type of the credential definition id
             const schemaId = credentialDefinitionId.startsWith('did:indy')
-                ? (0, identifiers_1.getDidIndySchemaId)(pool.indyNamespace, namespaceIdentifier, schema.schema.name, schema.schema.version)
+                ? (0, identifiers_1.getDidIndySchemaId)(pool.indyNamespace, schema.schema.issuerId, schema.schema.name, schema.schema.version)
                 : schema.schema.schemaId;
             return {
                 credentialDefinitionId,
@@ -624,7 +624,7 @@ class IndyVdrAnonCredsRegistry {
         }
     }
     async fetchIndySchemaWithSeqNo(agentContext, seqNo, did) {
-        var _a, _b;
+        var _a, _b, _c;
         const indyVdrPoolService = agentContext.dependencyManager.resolve(pool_1.IndyVdrPoolService);
         const { pool } = await indyVdrPoolService.getPoolForDid(agentContext, did);
         agentContext.config.logger.debug(`Getting transaction with seqNo '${seqNo}' from ledger '${pool.indyNamespace}'`);
@@ -637,14 +637,15 @@ class IndyVdrAnonCredsRegistry {
             return null;
         }
         const schema = (_b = response.result.data) === null || _b === void 0 ? void 0 : _b.txn.data;
-        const schemaId = (0, anoncreds_1.getUnqualifiedSchemaId)(did, schema.data.name, schema.data.version);
+        const schemaDid = (_c = response.result.data) === null || _c === void 0 ? void 0 : _c.txn.metadata.from;
+        const schemaId = (0, anoncreds_1.getUnqualifiedSchemaId)(schemaDid, schema.data.name, schema.data.version);
         return {
             schema: {
                 schemaId,
                 attr_name: schema.data.attr_names,
                 name: schema.data.name,
                 version: schema.data.version,
-                issuerId: did,
+                issuerId: schemaDid,
                 seqNo,
             },
             indyNamespace: pool.indyNamespace,
